scripts/validate_rst_title_capitalization.py

```diff
@@ -0,0 +1,394 @@
+        elif (node.parent.parent.tagname == 'title' and
+                node.parent.tagname in listOfMarkers):
+            lineno = findLineNumber(node)
+            myText = myText + node.astext()
+            beforeMarker = True
+            markerGrandparent = node.parent.parent
+        # Case 3: Encounter parent tagname of none of the above (Ex. 'paragraph')
+        else:
+            beforeMarker = False
+            if (myText != ""):
+                titleList.append(myText)
+                lineNumberList.append(lineno)
+                myText = ""
+                lineno = 0
+
+    # Leftover string that hasn't been appended yet due to how the for loop works
+    if (myText != ""):
+        titleList.append(myText)
+        lineNumberList.append(lineno)
+
+    # Return a list of the headings and a list of their corresponding line numbers
+    return titleList, lineNumberList
+
+
+def fillBadTitleDictionary(rstFile: str) -> None:
+    '''
+    Method that prints all of the bad titles
+    Message: [directory of rstFile, line number of bad title, error message]
+
+    '''
+
+    # Ensure file isn't one that causes the code to crash
+    if rstFile in cannotValidate:
+        return
+
+    # Ensure this file doesn't already have a badtitleDictionary slot
+    if rstFile in badTitleDictionary:
+        return
+
+    # Parse rstFile with an RST parser
+    document = parseRST(rstFile)
+
+    # Make a list of headings along with their line numbers from document tree
+    titleList, lineNumberList = findBadTitlesInDoctree(document)
+
+    # Append the badTitleDictionary if the capitalization convention not followed
+    for i in range(len(titleList)):
+        if not followCapitalizationConvention(titleList[i]):
+            if rstFile not in badTitleDictionary:
+                badTitleDictionary[rstFile] = [(titleList[i], lineNumberList[i])]
+            else:
+                badTitleDictionary[rstFile].append((titleList[i], lineNumberList[i]))
+
+
+def createRSTDirectoryList(source_paths: List[str]) -> List[str]:
+    '''
+    Given the command line arguments of directory paths, this method
+    creates a list of all of the .rst file directories that these paths contain
+
+    '''
+    # List of .rst file paths
+    f = []
+
+    # Loop through source_paths, recursively looking for .rst files
+    for directoryAddress in source_paths:
+        if not os.path.exists(directoryAddress):
+            raise ValueError(
+                "Please enter a valid path, pointing to a valid file/directory."
+            )
+        elif (directoryAddress.endswith(".rst")):
+            f.append(directoryAddress)
+        else:
+            for (dirpath, dirnames, filenames) in walk(directoryAddress):
+                for file in filenames:
+                    if file.endswith(".rst"):
+                        f.append(os.path.join(dirpath, file))
+
+    # Return the filled up list of .rst file paths
+    return f
```

Comment:
I think it's probably better to use a generator for this. Also, I'd avoid making the function "create", I think it's misleading (like creating a directory in disk...)